src/main/java/org/springframework/data/elasticsearch/core/DefaultIndexOperations.java

```diff
@@ -261,8 +227,40 @@ protected void doRefresh(IndexCoordinates index) {
 			}
 			return aliasMetaDataList;
 		} catch (IOException e) {
-			throw new ElasticsearchException("Could not map alias response : " + aliasResponse, e);
+			throw new UncategorizedElasticsearchException("Could not map alias response : " + aliasResponse, e);
+		}
+	}
+	// endregion
+
+	// region clientcallback
+	public <T> T execute(ElasticsearchRestTemplate.ClientCallback<T> callback) throws RuntimeException {
```

Comment:
Can we make the `DefaultIndexOperations` to use the REST Template API? That way, we can reuse the client callback and exception translation without us duplicating the code.

Reply:
great idea; by doing this I can remove the RestClient from the `DefaultIndexOperations`